### PR TITLE
Fix code smells reported by SonarCloud

### PR DESCRIPTION
**Pull request by zcahana**
1. Extract duplicate magic strings as constants
2. Rename variables to adhere to naming conventions
3. Refactor methods with high cognitive complexity

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/hyperconverged/hyperconverged_controller.go [hunk 1]
```diff
@@ -361,6 +304,30 @@ func (r *ReconcileHyperConverged) doReconcile(req *common.HcoRequest) (reconcile
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileHyperConverged) constructHcoRequest(ctx context.Context, request reconcile.Request) (*common.HcoRequest, error) {
```
Comment (nunnatsa) [inline pkg/controller/hyperconverged/hyperconverged_controller.go:307]: This function does two things. It check if the source of the request is a change in HC or in a secondery watched resource, *and* creates the hcoRequest. The latest already implemented in common.NewHcoRequest.

I think this method need a new name and the last line should be placed in the original place, out of this method.

Comment (zcahana) [reply pkg/controller/hyperconverged/hyperconverged_controller.go]: Done.

File: pkg/controller/hyperconverged/hyperconverged_controller.go [hunk 2]
```diff
@@ -615,52 +565,48 @@ func (r *ReconcileHyperConverged) aggregateComponentConditions(req *common.HcoRe
...
 
-			if _, conditionFound = req.Conditions[conditionsv1.ConditionAvailable]; !conditionFound { // (#chart 17) {
-				req.Conditions.SetStatusCondition(conditionsv1.Condition{ // (#chart 18)
-					Type:    conditionsv1.ConditionAvailable,
-					Status:  corev1.ConditionTrue,
-					Reason:  reconcileCompleted,
-					Message: reconcileCompletedMessage,
-				})
-			}
 		}
```
Comment (nunnatsa) [inline pkg/controller/hyperconverged/hyperconverged_controller.go:607]: Worth a new function?

File: pkg/controller/hyperconverged/hyperconverged_controller.go [hunk 3]
```diff
@@ -790,6 +736,55 @@ func (r *ReconcileHyperConverged) detectTaintedConfiguration(req *common.HcoRequ
 	}
 }
 
+func (r *ReconcileHyperConverged) updateHCO(req *common.HcoRequest) error {
```
Comment (nunnatsa) [inline pkg/controller/hyperconverged/hyperconverged_controller.go:739]: we're not updating HCO - HC operator, but HC CR. Pleasse consider renaming.

**Review by nunnatsa — COMMENTED** [review]
Added some low priority comments. maybe for next time? Not blocking changes.

File: pkg/controller/hyperconverged/hyperconverged_controller.go (hunk 3 above)
Comment (zcahana) [reply pkg/controller/hyperconverged/hyperconverged_controller.go:739]: Good point. Will rename.

File: pkg/controller/hyperconverged/hyperconverged_controller.go (hunk 2 above)
Comment (zcahana) [reply pkg/controller/hyperconverged/hyperconverged_controller.go:607]: Considered this (regarding these 3 comments) - I think it'll cause the overall flow to become less clear than is now.

**Comment by zcahana** [comment]
Some suspicious failures in the different lanes here - need to investigate what's causing them.

**Comment by zcahana** [comment]
Some lanes are failing due to timeouts after long hours of test execution - not clear where it's stuck.
Temporarily reverting 9f58a27 while trying to locate the offending code.

**Comment by zcahana** [comment]
Tests pass after reverting 38b94c0. Hence, this is the offending commit. Reason is not clear yet - debugging...

**Review by nunnatsa — APPROVED** [review]
/approve

**Pull request merged**